### PR TITLE
Update message handler to handle the priority data

### PR DESCRIPTION
**Pull request by tcar121293**
Update message handler to handle the priority data
## Description
Update erc20 and erc721 message handlers
Add unit tests

## Related Issue Or Context

Closes: #235 

## How Has This Been Tested? Testing details.
Added unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/evm/voter/message-handler.go [hunk 1]
```diff
@@ -70,40 +71,52 @@ func (mh *EVMMessageHandler) RegisterMessageHandler(address string, handler Mess
...
 	data = append(data, common.LeftPadBytes(recipientLen, 32)...) // length of recipient (uint256)
 	data = append(data, recipient...)                             // recipient ([]byte)
+
+	if payloadLen == 3 {
+		priority, ok := m.Payload[2].([]byte)
+		if !ok {
+			return nil, errors.New("wrong payload priority format")
+		}
+		priorityLen := big.NewInt(int64(len(priority))).Bytes()
+		data = append(data, common.LeftPadBytes(priorityLen, 1)...) // length of priority (uint8)
```
Comment (mpetrun5) [inline chains/evm/voter/message-handler.go:98]: Does data have to contain priority data? Seems irrelevant for submitting on chain.. 
I would probably add https://www.sohamkamani.com/golang/options-pattern/ (`...options`) param to creating new proposal and add all extra options that just the relayer needs.

File: chains/evm/voter/message-handler.go [hunk 2]
```diff
@@ -70,40 +71,52 @@ func (mh *EVMMessageHandler) RegisterMessageHandler(address string, handler Mess
...
+		return nil, errors.New("wrong payload recipient format")
 	}
 	var data []byte
 	data = append(data, common.LeftPadBytes(amount, 32)...) // amount (uint256)
 	recipientLen := big.NewInt(int64(len(recipient))).Bytes()
 	data = append(data, common.LeftPadBytes(recipientLen, 32)...) // length of recipient (uint256)
 	data = append(data, recipient...)                             // recipient ([]byte)
+
+	if payloadLen == 3 {
+		priority, ok := m.Payload[2].([]byte)
```
Comment (mpetrun5) [inline chains/evm/voter/message-handler.go:93]: Probably move logic for extracting priority to separate function.

Comment (mpetrun5) [reply chains/evm/voter/message-handler.go:93]: Also, why does priority have to be in payload? If we extract priority in event handler into some message metadata that can contain priority and other data then we don't have to have logic here to check if payload is malformed and also we can just pass it to NewProposal as is.

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: chains/evm/voter/message-handler.go [hunk 3]
```diff
@@ -113,7 +116,8 @@ func ERC721MessageHandler(msg *message.Message, handlerAddr, bridgeAddress commo
 	metadataLen := big.NewInt(int64(len(metadata))).Bytes()
 	data.Write(common.LeftPadBytes(metadataLen, 32))
 	data.Write(metadata)
-	return proposal.NewProposal(msg.Source, msg.DepositNonce, msg.ResourceId, data.Bytes(), handlerAddr, bridgeAddress), nil
+
```
Comment (P1sar) [inline chains/evm/voter/message-handler.go:119]: I do not think Go-style suppose to have spaces before return statement

File: chains/evm/voter/message-handler.go [hunk 4]
```diff
@@ -122,11 +126,12 @@ func GenericMessageHandler(msg *message.Message, handlerAddr, bridgeAddress comm
...
 	if !ok {
-		return nil, errors.New("unable to convert metadata to []byte")
+		return nil, errors.New("wrong payload metadata format")
 	}
 	data := bytes.Buffer{}
 	metadataLen := big.NewInt(int64(len(metadata))).Bytes()
 	data.Write(common.LeftPadBytes(metadataLen, 32)) // length of metadata (uint256)
 	data.Write(metadata)
-	return proposal.NewProposal(msg.Source, msg.DepositNonce, msg.ResourceId, data.Bytes(), handlerAddr, bridgeAddress), nil
+
```
Comment (P1sar) [inline chains/evm/voter/message-handler.go:135]: same as above

**Review by P1sar — CHANGES_REQUESTED** [review]

**Review by MakMuftic — APPROVED** [review]
LGTM

**Review by nmlinaric — APPROVED** [review]

**Review by P1sar — APPROVED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**